homeassistant/components/tplink/common.py

```diff
@@ -58,13 +64,100 @@ def has_device_with_host(self, host: str) -> bool:
         return False
 
 
+class Discover(BaseDiscover):
```

Comment:
This looks like protocol level info that should be part of the device library instead of Home Assistant.

Reply:
I agree, and it's in fact taken from an upstream change to the pyHS100 library. But that change is unreleased and the library is now abandoned by @rytilahti . The only other solution I see is for me to fork the library, change it there and point the manifest at the fork (difficult without pypi access though).

@rytilahti appears to be working on a new library but I haven't seen any HA integration work for it yet.